src/conn/conn_compact.c

```diff
@@ -484,6 +496,10 @@ __background_compact_server(void *arg)
         __wt_spin_lock(session, &conn->background_compact.lock);
         running = conn->background_compact.running;
         if (conn->background_compact.signalled) {
+            /* If configured to run once, start from the beginning. */
+            if (running && conn->background_compact.run_once)
```

Comment:
Is it guaranteed that we should be running at this point?

Reply:
Yes, you have to enable the background compaction server to set this setting.

Reply:
As discussed, remove the `running` flag and add an assert.